src/mainviews/list-jobs.tsx

```diff
@@ -148,38 +191,77 @@ export function NotebookJobsListBody(
     }
   ];
 
+  const rows: JSX.Element[] = notebookJobs.jobs
+    .filter(job => !deletedRows.has(job.job_id))
+    .map(job =>
+      buildTableRow(
+        job,
+        props.app,
+        props.showCreateJob,
+        deleteRow,
+        translateStatus
+      )
+    );
+
+  const handlePageChange = async (e: unknown, newPage: number) => {
+    // if newPage <= maxPage, no need to fetch more rows
+    if (newPage <= maxPage) {
+      setPage(newPage);
+      return;
+    }
+
+    await fetchMoreRows(notebookJobs.next_token);
+    setPage(newPage);
+    setMaxPage(newPage);
+  };
+
+  // note that the parent must be a JSX fragment for DataGrid to be sized properly
   return (
     <>
       {reloadButton}
-      <div className={`${ListItemClass} jp-notebook-job-list-header`}>
-        {columns.map((column, idx) => (
-          <NotebookJobsColumnHeader
-            key={idx}
-            gridColumn={column}
-            jobsQuery={jobsQuery}
-            setJobsQuery={setJobsQuery}
+      {/* outer div expands to fill rest of screen */}
+      <div style={{ flex: 1, height: 0 }}>
+        <TableContainer
+          component={Paper}
+          sx={{
+            height: '100%',
+            ...(loading ? { pointerEvents: 'none', opacity: 0.5 } : {})
+          }}
+        >
+          <Table stickyHeader>
+            <TableHead>
+              {columns.map((column, idx) => (
+                <NotebookJobsColumnHeader
+                  key={idx}
+                  gridColumn={column}
+                  jobsQuery={jobsQuery}
+                  setJobsQuery={setJobsQuery}
+                />
+              ))}
+            </TableHead>
+            <TableBody>
+              {rows.slice(page * PAGE_SIZE, (page + 1) * PAGE_SIZE)}
+            </TableBody>
+          </Table>
+          <TablePagination
+            component="div"
+            sx={{
+              position: 'sticky',
+              bottom: 0,
+              backgroundColor: 'white',
+              borderTop: '1px solid #e0e0e0'
```

Comment:
These will not work with themes. We should find a way to use the JL theme vars here.

Reply:
Thank you. Very observant.